src/utransport.rs

```diff
@@ -238,15 +238,15 @@ pub trait UTransport {
     /// let my_listener: Arc<dyn UListener> = Arc::new(MyListener::new());
     /// // ...send a clone through when registering...
     /// let my_listener_for_transport = my_listener.clone();
-    /// let register_result = my_transport.register_listener(my_uuri.clone(), &my_listener_for_transport);
+    /// let register_result = my_transport.register_listener(my_uuri.clone(), my_listener_for_transport.clone());
```

Comment:
`my_listener_for_transport` already is a clone of the original listener ...
FMPOV this should simply be
```rust
let my_listener: Arc<dyn UListener> = Arc::new(MyListener::new());
let register_result = my_transport.register_listener(my_uuri.clone(), my_listener.clone());
// ...and use the original we hung onto when unregistering
let unregister_result = my_transport.unregister_listener(my_uuri, my_listener);
```

Reply:
Ahhh, you're right. Looking at it now it's obvious. Thanks for catching this